examples/platform/nxp/se05x/DeviceAttestationSe05xCredsExample_v2.cpp

```diff
@@ -0,0 +1,298 @@
+                                CHIP_ERROR_INTERNAL);
+        }
+        /* Set time stamp length */
+        VerifyOrReturnError(CHIP_NO_ERROR == se05xSetCertificate(CD_TIME_STAMP_LEN_SE05X_ID, &tslen, 1), CHIP_ERROR_INTERNAL);
+
+        if ((tagReader.GetRemainingLength() + 1 /* End container */) >= 16)
+        {
+            /* Set attestation challenge */
+            VerifyOrReturnError(CHIP_NO_ERROR ==
+                                    se05xSetCertificate(CD_ATTEST_CHALLENGE_SE05X_ID, (message_to_sign.end() - 16), 16),
+                                CHIP_ERROR_INTERNAL);
+        }
+    }
+    else
+    {
+        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(1), tagReader));
+        uint8_t csrlen = tagReader.GetLength();
+        VerifyOrReturnError(csrlen > 0, CHIP_ERROR_INVALID_TLV_TAG);
+        ByteSpan csr_data;
+        /* Get nocsr */
+        ReturnErrorOnFailure(tagReader.Get(csr_data));
+        /* Set nocsr length */
+        VerifyOrReturnError(CHIP_NO_ERROR == se05xSetCertificate(NOCSR_CSR_LEN_SE05X_ID, &csrlen, 1), CHIP_ERROR_INTERNAL);
+        /* Set nocsr data */
+        se05x_delete_key(NOCSR_CSR_DATA_SE05X_ID);
+        VerifyOrReturnError(CHIP_NO_ERROR == se05xSetCertificate(NOCSR_CSR_DATA_SE05X_ID, csr_data.data(), csr_data.size()),
+                            CHIP_ERROR_INTERNAL);
+
+        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(2), tagReader));
+        uint8_t noncelen = tagReader.GetLength();
+        VerifyOrReturnError(noncelen > 0, CHIP_ERROR_INVALID_TLV_TAG);
+        /* Get nocsr nonce */
+        ByteSpan nocsr_nonce;
+        ReturnErrorOnFailure(tagReader.Get(nocsr_nonce));
+        /* Set nocsr nonce data */
+        VerifyOrReturnError(CHIP_NO_ERROR ==
+                                se05xSetCertificate(NOCSR_CSR_NONCE_DATA_SE05X_ID, nocsr_nonce.data(), nocsr_nonce.size()),
+                            CHIP_ERROR_INTERNAL);
+
+        if ((tagReader.GetRemainingLength() + 1 /* End container */) >= 16)
+        {
+            /* Set attestation challenge */
+            VerifyOrReturnError(CHIP_NO_ERROR ==
+                                    se05xSetCertificate(NOCSR_ATTEST_CHALLENGE_SE05X_ID, (message_to_sign.end() - 16), 16),
+                                CHIP_ERROR_INTERNAL);
+        }
+    }
+
+    uint8_t signature_se05x[Crypto::kMax_ECDSA_Signature_Length_Der] = { 0 };
+    size_t signature_se05x_len                                       = sizeof(signature_se05x);
+
+    if (sign_cert_decl_attest)
+    {
+        err = se05xPerformInternalSign(CD_DEV_ATTESTATION_KEY_SE05X_ID, signature_se05x, &signature_se05x_len);
+        se05x_delete_key(CD_ATTEST_NONCE_DATA_SE05X_ID);
+        se05x_delete_key(CD_TIME_STAMP_LEN_SE05X_ID);
+        se05x_delete_key(CD_TIME_STAMP_DATA_SE05X_ID);
+        se05x_delete_key(CD_ATTEST_CHALLENGE_SE05X_ID);
+        sign_cert_decl_attest = 0;
```

Comment:
This setup with sign_cert_decl_attest needs to be very clearly documented, especially because it's not spec-compliant....